pkg/rbd/nodeserver.go

```diff
@@ -42,12 +40,23 @@ type NodeServer struct {
 }
 
 // NodeStageVolume mounts the volume to a staging path on the node.
+// Implementation notes:
+// - stagingTargetPath is the directory passed in the request where the volume needs to be staged
```

Comment:
why do we need to create directory and files based on the block/file?

can we just create a file inside the stagingPath and use it?

Reply:
Target of a flesystem mount needs to be a directory, and we need the target as a file when bind mounting a devoce for the block use-case. Hence the difference.

Reply:
cant we just create a `.config.json` inside a stagingTargetPath directory and use it?  why create directory and file?

Reply:
The comment reads as below:

// - stagingTargetPath is the directory passed in the request where the volume needs to be staged
//   - We stage the volume into a directory, named after the VolumeID inside stagingTargetPath if
//    it is a file system
//   - We stage the volume into a file, named after the VolumeID inside stagingTargetPath if it is
//    a block volume

So, we mount the device on a directory inside the `stagingTargetPath`, the directory is named based on the `VolumeID` OR we bind mount the device to a file named similarly.

We create and maintain the `config.json` inside the 'stagingTargetPath` which is outside either of these 2 mounts and hence outside the mount point and accessible even post the umount etc.

So to answer your question, `config.json` is created just within the passed in `stagingTargetPath`, the file/directory is created to stage the image as required.

I do not see how not creating a sub-entity where we stage can be avoided if that is your direction of thought.

Reply:
if `stagingTargetPath` looks like `/var/lib/kubelet/kubernetes.io/pv/pvc-xxxx-xxxx-xxx/globalmount` can we just create `/var/lib/kubelet/kubernetes.io/pv/pvc-xxxx-xxxx-xxx/.config.json` and use it?

this will be available for unmapping also

Reply:
I would not attempt to use a path outside the scope of what is passed to the plugin, IOW something akin to `cd ..`  and then create files in there. As this assumes we have rights to manage files and directories outside the passed in `stagingTargetPath` which I believe is breaking the abstraction provided.

So even though we know what the path is, I would surely not want to do this.